[BUG] NPE in alluxio runtime

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

```
2021-04-04T08:39:32.630+0800	INFO	alluxioctl.AlluxioRuntime	alluxio/runtime_info.go:42	Dataset is notfound	{"alluxioruntime": "default/hbase", "name": "hbase", "namespace": "default"}
E0404 08:39:32.630672       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 327 [running]:
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14cf0a0, 0x24fe0c0)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14cf0a0, 0x24fe0c0)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/fluid-cloudnative/fluid/api/v1alpha1.(*Dataset).GetObjectKind(0x0, 0xc000b2ad4f, 0x1)
	<autogenerated>:1 +0x5
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.kubeObjectWrapper.MarshalLogObject(0x189b380, 0x0, 0x18d8ea0, 0xc000620420, 0x203000, 0x40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:67 +0x49
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AppendObject(0xc000620420, 0x7f57bf83af88, 0xc000348490, 0x700, 0x7f57bf83af88)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:192 +0x99
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AddObject(0xc000620420, 0x168dd69, 0x7, 0x7f57bf83af88, 0xc000348490, 0xc000b2af18, 0x47d480)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:99 +0x61
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.Field.AddTo(0x168dd69, 0x7, 0x2, 0x0, 0x0, 0x0, 0x15274c0, 0xc000348490, 0x18d8ea0, 0xc000620420)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:114 +0x242
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.addFields(0x18d8ea0, 0xc000620420, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:199 +0xcf
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.writeContext(0xc000171da0, 0xc000116fc0, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:131 +0xc5
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.EncodeEntry(0xc000171da0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:110 +0x3d6
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.(*KubeAwareEncoder).EncodeEntry(0xc00019cfe0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:128 +0x174
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*ioCore).Write(0xc000171dd0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/core.go:86 +0xa9
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0000fcbb0, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/entry.go:215 +0x117
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*infoLogger).Info(0xc00019d008, 0x1697f42, 0x12, 0xc000ba2a20, 0x2, 0x2)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:70 +0xdd
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio.(*AlluxioEngine).AssignNodesToCache(0xc00014c0a0, 0xc000000002, 0x0, 0x0, 0x7)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/node.go:32 +0x14a
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio.(*AlluxioEngine).SetupWorkers(0xc00014c0a0, 0x1, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/worker.go:43 +0x78
github.com/fluid-cloudnative/fluid/pkg/ddc/base.(*TemplateEngine).Setup(0xc0003c2000, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, 0xc000aaa000, 0xc001fb1d40, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/base/setup.go:79 +0x446
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileRuntime(0xc0002c8900, 0x18c3c20, 0xc0003c2000, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:215 +0x58b
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal(0xc0002c8900, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, 0xc000aaa000, 0xc001fb1d40, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:135 +0x5a0
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio.(*RuntimeReconciler).Reconcile(0xc00034e8e0, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x0, 0xc012609043f801ce, 0xc0000f2480, 0xc0000f23f8)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio/alluxio_runtime_controller.go:97 +0x715
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00071e540, 0x1527560, 0xc0008b4600, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x161
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00071e540, 0x203000)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xae
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00071e540)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000a91850)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000a91850, 0x1881100, 0xc00003bf50, 0x1, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a91850, 0x3b9aca00, 0x0, 0x1, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000a91850, 0x3b9aca00, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x305
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12929b5]

goroutine 327 [running]:
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x14cf0a0, 0x24fe0c0)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/fluid-cloudnative/fluid/api/v1alpha1.(*Dataset).GetObjectKind(0x0, 0xc000b2ad4f, 0x1)
	<autogenerated>:1 +0x5
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.kubeObjectWrapper.MarshalLogObject(0x189b380, 0x0, 0x18d8ea0, 0xc000620420, 0x203000, 0x40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:67 +0x49
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AppendObject(0xc000620420, 0x7f57bf83af88, 0xc000348490, 0x700, 0x7f57bf83af88)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:192 +0x99
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AddObject(0xc000620420, 0x168dd69, 0x7, 0x7f57bf83af88, 0xc000348490, 0xc000b2af18, 0x47d480)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:99 +0x61
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.Field.AddTo(0x168dd69, 0x7, 0x2, 0x0, 0x0, 0x0, 0x15274c0, 0xc000348490, 0x18d8ea0, 0xc000620420)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:114 +0x242
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.addFields(0x18d8ea0, 0xc000620420, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:199 +0xcf
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.writeContext(0xc000171da0, 0xc000116fc0, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:131 +0xc5
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.EncodeEntry(0xc000171da0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:110 +0x3d6
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.(*KubeAwareEncoder).EncodeEntry(0xc00019cfe0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:128 +0x174
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*ioCore).Write(0xc000171dd0, 0x0, 0xc0126091259504b8, 0xc9b500310fa1, 0x251a580, 0xc0003788a0, 0x19, 0x1697f42, 0x12, 0x1, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/core.go:86 +0xa9
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0000fcbb0, 0xc000020480, 0x1, 0x1)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/entry.go:215 +0x117
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*infoLogger).Info(0xc00019d008, 0x1697f42, 0x12, 0xc000ba2a20, 0x2, 0x2)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:70 +0xdd
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio.(*AlluxioEngine).AssignNodesToCache(0xc00014c0a0, 0xc000000002, 0x0, 0x0, 0x7)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/node.go:32 +0x14a
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio.(*AlluxioEngine).SetupWorkers(0xc00014c0a0, 0x1, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/worker.go:43 +0x78
github.com/fluid-cloudnative/fluid/pkg/ddc/base.(*TemplateEngine).Setup(0xc0003c2000, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, 0xc000aaa000, 0xc001fb1d40, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/base/setup.go:79 +0x446
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileRuntime(0xc0002c8900, 0x18c3c20, 0xc0003c2000, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:215 +0x58b
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal(0xc0002c8900, 0x18bd040, 0xc0001a6010, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x18c3ce0, 0xc000aaa000, 0xc001fb1d40, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:135 +0x5a0
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio.(*RuntimeReconciler).Reconcile(0xc00034e8e0, 0xc000172b90, 0x7, 0xc000172b80, 0x5, 0x0, 0xc012609043f801ce, 0xc0000f2480, 0xc0000f23f8)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio/alluxio_runtime_controller.go:97 +0x715
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00071e540, 0x1527560, 0xc0008b4600, 0x0)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x161
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00071e540, 0x203000)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xae
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00071e540)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000a91850)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000a91850, 0x1881100, 0xc00003bf50, 0x1, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a91850, 0x3b9aca00, 0x0, 0x1, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000a91850, 0x3b9aca00, 0xc000028b40)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x305
Get K8S used ports, [0 0 0 11270 0 9100 0 0 9100 0 11270 9100 9100 0 0 0 11260 0 0 0 0 0 0 0 0 0 32009 31214]%
```

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

--- a/pkg/ddc/alluxio/node.go
+++ b/pkg/ddc/alluxio/node.go
@@ -29,11 +29,11 @@ func (e *AlluxioEngine) AssignNodesToCache(desiredNum int32) (currentScheduleNum
 	}
 
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
-	e.Log.Info("AssignNodesToCache", "dataset", dataset)
 	if err != nil {
 		return
 	}
 
+	e.Log.Info("AssignNodesToCache", "dataset", dataset)
 	return datasetSchedule.AssignDatasetToNodes(runtimeInfo,
 		dataset,
 		e.Client,

